packages/devtools_server/lib/src/server.dart

```diff
@@ -554,10 +555,10 @@ File _devToolsBackup;
 
 bool backupAndCreateDevToolsStore() {
   assert(_devToolsBackup == null);
-  final devToolsStore = File('${DevToolsUsage.userHomeDir()}/.devtools');
+  final devToolsStore = File('${LocalFileSystem.userHomeDir()}/.devtools');
```

Comment:
rather than exposing userHomeDIr which is a terrifying thing to allow access to paths relative to, what if we expose only
`devtoolsRootDir()` which can be `userHomeDir()/.devtools`
That will help avoid cases where these server APIs could be abused to access arbitrary files on the user's machine.

Reply:
We'll still need userHomeDir to access other files but we should never use it for file serving logic due to security concerns.